components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -1209,33 +1218,48 @@ private String getClaimValue(String userId, AbstractUserStoreManager userStoreMa
         }
     }
 
-    private void setStepConfigAuthenticatorList(SequenceConfig effectiveSequence) {
+    /**
+     * The organization SSO IDP is added for Console & MyAccount to provide sub-organization access. But the
+     * organization SSO IDP should be hidden from the login sequence unless the organization SSO option is requested via
+     * the fidp parameter.
+     *
+     * @param effectiveSequence The effective sequences.
+     */
+    private void removeOrganizationSsoStepsForPortalApps(SequenceConfig effectiveSequence) {
 
         Map<Integer, StepConfig> stepMap = effectiveSequence.getStepMap();
-        ApplicationConfig applicationConfig = effectiveSequence.getApplicationConfig();
-        if (MapUtils.isEmpty(stepMap) && applicationConfig == null) {
-            return;
-        }
-        ServiceProviderProperty[] spProperties = applicationConfig.getServiceProvider().getSpProperties();
-        boolean showOrganizationAuthenticator = true;
-        for (ServiceProviderProperty property : spProperties) {
-            if (IS_APP_SHARED.equals(property.getName()) && !Boolean.parseBoolean(property.getValue())) {
-                showOrganizationAuthenticator = false;
-                break;
-            }
```

Comment:
Isn't that removing this logic affect to the behavior of other shared apps?

Reply:
The logic should be applied for the console & myaccount. Other B2B apps should get organization SSO removed from the login options when there is no sub-organizations exist. 
Will check for the b2b apps whether organization SSO is removed for the following cases.
1 - Unshare with all orgs.
2 - Delete all the sub-orgs
3 - Unshare with on org which is the last app which is shared.